### ISSUE TITLE
introPrice.periodNumberOfUnits is returning 7 on iOS

### ISSUE THREAD

**Issue by LydGol90** [RevenueCat/react-native-purchases#348]
- [X ] I have updated Purchases SDK to the latest version
- [X ] I have read the [Contribution Guidelines](https://github.com/RevenueCat/react-native-purchases/blob/main/CONTRIBUTING.md)
- [X ] I have searched the [Community](https://community.revenuecat.com)
- [X ] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [ X] I have searched for [existing Github issues](https://github.com/RevenueCat/react-native-purchases/issues)

Hi there. At some point in mid April it looks like we started sometimes getting a value of 7 under `introPrice.periodNumberOfUnits` on our current product when calling `getOfferings` on iOS. This product in app store has a 1 week free trial. Sometimes it is still 1 but randomly switches to 7. Since Apple only offer a distinct set of options for introductory offers, none of which should have `periodNumberOfUnits` as 7, we are slightly confused as to how this can be happening. We have checked everything on our side and we are definitely not fiddling with this value or hardcoding any values, we just display whatever number is returned under that value, and we also send it to Segment in some of our tracking. The issue also doesn't correlate with a particular app release on our side but is instead affecting all versions. The issue is quite concerning as it means that sometimes our users are incorrectly seeing that they have a 7-week free trial advertised in the app.

1. Environment
   1. Platform: iOS
   2. SDK version: 4.5.2 (react-native-purchases)
   3. OS version: multiple
   4. Xcode/Android Studio version:
   5. React Native version: 0.68.0
   6. SDK installation (CocoaPods + version or manual): cocoapods
   7. How widespread is the issue. Percentage of devices affected: From data it looks like in about 20% of cases we are getting a 7 instead of a 1 when fetching this value.

Thanks in advance for your help,
Lydia

**Comment by vegaro** [comment]
Hi @LydGol90 , do you mind sharing the other fields of `introPrice`? I would like to see how the period, cycles and unit fields look like.

Thanks!

**Comment by aboedo** [comment]
The fact that it's oscillating seems really odd and is probably an Apple bug. Maybe it's going back and forth between `1 week` and `7 days`?

**Comment by vegaro** [comment]
Yes, that's my thought (which is still weird), but my thought is that maybe it says 7 period units and unit is DAY?

**Comment by LydGol90** [comment]
> Hi @LydGol90 , do you mind sharing the other fields of `introPrice`? I would like to see how the period, cycles and unit fields look like.
> 
> Thanks!

@vegaro Sorry missed this on first read. I will try and see if I can get these values. Issue is this is on our production app and I don't think we bother sending those values to our tracking but I will check

**Comment by LydGol90** [comment]
@vegaro Luckily we do send them!

7 day
periodUnit: DAY
periodNumberOfUnits: 7
period: P7D
cycles: 1

1 week
periodUnit: WEEK
periodNumberOfUnits: 1
period: P1W
cycles: 1

**Comment by LydGol90** [comment]
@vegaro Did you get a chance to investigate this? Thanks

**Comment by vegaro** [comment]
@LydGol90 sorry for the late reply

I looked into it a bit. We map directly what iOS sends us so for some reason, StoreKit is sending  an [introductory price ](https://developer.apple.com/documentation/storekit/skproduct/2936878-introductoryprice) with a period of 7 days and sometimes with a period of 1 week.

Some ideas why this could be happening:
- Maybe it depends on iOS version? Can you check if the iOS versions differ? I am not sure if you have access to that information.
- Do you have an intro price configured apart of a free trial? Maybe StoreKit is being inconsistent about how it sends the periods between intro price types.

We are going to file a radar with Apple about this, maybe they have some insights about it.

Another thing is that you mention:

> The issue is quite concerning as it means that sometimes our users are incorrectly seeing that they have a **7-week free trial** advertised in the app.

Please note that we are sending either 7 days or 1 week, which is the same. Not 7 weeks.

**Comment by LydGol90** [comment]
Hi @vegaro 

Thank you!

I don't think it can be related to version because I am just opening the app on the same phone and it switches from 7-day to 1-week between opens

We only have one introductory price on that product, which is the 1 week free trial. We do have another product which has a 3 day free trial but that is a whole separate product and is 3 days not 7. That one is working correctly.

I find it weird because apple don't even offer a 7 day free trial so no idea how they are returning it!

On the last point, you are right, that issue was one our end. You are right we only get 1 week or 7 day.

Thanks

**Comment by vegaro** [comment]
Thanks for your reply. Are you running the app in a simulator? Or is it the app you have in production and downloaded from App Store? Just curious if this could be related to StoreKit config files.

I'm going to create a ticket on our end to see if there's anything we can do. Maybe we can normalize 7 days to 1 week, 14 days to 2 weeks, 4 weeks to 1 month, etc. Just to be clear, it is low priority for us right now, so I'm not sure when we will get to it.

I hope that makes sense.

**Comment by LydGol90** [comment]
@vegaro Nope this is on our production app downloaded from the app store, as well as the one downloaded from Testflight. Yes I understand that it isn't a high priority, mostly just wanted to understand what is causing it, but if it isn't an issue on your end then not much you can do. Thanks

**Comment by fire-at-will** [comment]
@vegaro @LydGol90 I just merged [purchases-ios PR #1591](https://github.com/RevenueCat/purchases-ios/pull/1591) which addresses this issue from StoreKit! This should resolve the issue once it gets released in the next version of `purchases-ios` and adopted in a future version `react-native-purchases`.

**Comment by vegaro** [comment]
Thanks @fire-at-will . We will release the fix in the next major of the plugin

**Comment by aboedo** [comment]
Happy to report that the fix has been released as part of 5.0.0! 🎉
